mysqlnd_azure.c

```diff
@@ -534,35 +646,56 @@ MYSQLND_METHOD(mysqlnd_azure, connect)(MYSQLND * conn_handle,
 			mysqlnd_options4(conn_handle, MYSQL_OPT_CONNECT_ATTR_ADD, "_server_host", hostname.s);
 		}
 
-		if(!MYSQLND_AZURE_G(enabled)) {
+		if (!MYSQLND_AZURE_G(enabled)) {
 			DBG_ENTER("mysqlnd_azure::connect redirect disabled");
 			ret = org_conn_d_m.connect(*pconn, hostname, username, password, database, port, socket_or_pipe, mysql_flags);
 		}
 		else {
 			DBG_ENTER("mysqlnd_azure::connect redirect enabled");
 
-			//first check whether the redirect info already cached
-			MYSQLND_AZURE_REDIRECT_INFO* redirect_info = mysqlnd_azure_find_redirect_cache(*pconn, username.s, hostname.s, port);
-			if (redirect_info != NULL) {
-				DBG_ENTER("mysqlnd_azure::connect try the cached info first");
-				MYSQLND_CSTRING redirect_host = { redirect_info->redirect_host, strlen(redirect_info->redirect_host) };
-				MYSQLND_CSTRING redirect_user = { redirect_info->redirect_user, strlen(redirect_info->redirect_user) };
-
-				mysqlnd_azure_set_is_using_redirect(*pconn, 1);
-				ret = (*pconn)->m->connect(pconn, redirect_host, redirect_user, password, database, redirect_info->redirect_port, socket_or_pipe, mysql_flags);
-				if (ret == FAIL) {
-					//remove invalid cache
-					mysqlnd_azure_remove_redirect_cache(*pconn, username.s, hostname.s, port);
-
-					mysqlnd_azure_set_is_using_redirect(*pconn, 0);
+			//Redirection is only possible with SSL at present. Continue with no redirection if SSL is not set
+			unsigned int temp_flags = (*pconn)->m->get_updated_connect_flags(*pconn, mysql_flags);
+			if (!(temp_flags & CLIENT_SSL)) {
+				ret = org_conn_d_m.connect(*pconn, hostname, username, password, database, port, socket_or_pipe, mysql_flags);
+			}
+			else { //SSL is enabled
+
+				//first check whether the redirect info already cached
+				MYSQLND_AZURE_REDIRECT_INFO* redirect_info = mysqlnd_azure_find_redirect_cache(username.s, hostname.s, port);
+				if (redirect_info != NULL) {
+					DBG_ENTER("mysqlnd_azure::connect try the cached info first");
+					const MYSQLND_CSTRING redirect_host = { redirect_info->redirect_host, strlen(redirect_info->redirect_host) };
+					const MYSQLND_CSTRING redirect_user = { redirect_info->redirect_user, strlen(redirect_info->redirect_user) };
+
+					MYSQLND_AZURE_CONN_DATA** data_ret = mysqlnd_azure_set_is_using_redirect(*pconn, 1);
+                    if(data_ret==NULL) {
+                        SET_OOM_ERROR((*pconn)->error_info);
+                        return FAIL;
+                    }
+
+					ret = (*pconn)->m->connect(pconn, redirect_host, redirect_user, password, database, redirect_info->redirect_port, socket_or_pipe, mysql_flags);
```

Comment:
Remember to format the code (tab) in the future

Reply:
About the format, I think it maybe editor related. It looks with right indent with my editor...